data-access-analysis-tools/adql-recipes.rst

```diff
@@ -0,0 +1,222 @@
+
+
+############
+ADQL Recipes
+############
+
+.. This section should provide a brief, top-level description of the page.
+
+ADQL is the `Astronomical Data Query Language <https://www.ivoa.net/documents/ADQL/>`_.
+The language is used by the `IVOA <https://ivoa.net>`_ to represent astronomy queries posted to Virtual Observatory (VO) services, such as the Rubin LSST Table Access Protocol (TAP) service.
+ADQL is based on the Structured Query Language (SQL).
+
+ADQL can be used in both the Notebook and Portal aspects:
+ - :ref:`Data-Access-Analysis-Tools-TAP-NB`
+ - how to :ref:`Portal-Intro-ADQL-Queries` in the Portal aspect
+
+Learn more about the `TAP-accessible DP0.2 catalogs <https://dp0-2.lsst.io/data-products-dp0-2/index.html#catalogs>`__.
+
+.. Important::
+    If a query takes longer than you expect, please submit a `GitHub Issue <https://github.com/rubin-dp0/Support>`__
+    or post in the "DP0 RSP Service Issues" category of the Rubin Community Forum.
+    Staff with familiarity with Qserv are happy to investigate and to help tweak queries for optimal execution.
+
+
+.. _Adql-Recipes-General-Advice:
+
+General Advice
+==============
+
+LSST Query Services (Qserv) provides access to the LSST Database Catalogs.
+Users can query the catalogs using standard SQL query language with a few `restrictions <https://qserv.lsst.io/user/index.html#restrictions>`__.
+
+**Fast queries:** 
+Qserv stores catalog data sharded by coordinate (RA, Dec).
+ADQL query statements that include constraints by coordinate do not requre a whole-catalog search,
+and are typically faster (and can be *much* faster) than ADQL query statements which only include constraints for other columns.
+
+**Recommended constraint:**
+Include ``detect_isPrimary = True`` in queries for the ``Object``, ``Source``, and ``ForcedSource`` catalogs.
+This parameter is ``True`` if a source has no children, is in the inner region of a coadd patch, is in the inner region of a coadd tract, and is not detected in a pseudo-filter.
+Including this constraint will remove any duplicates (i.e., will not include *both* a parent and its deblended children).
+
+Additional external resources for learning about SQL, ADQL, and Qserv include:
+ - `W3 School's SQL Tutorial <https://www.w3schools.com/sql/default.asp>`__
+ - `IVOA's ADQL Documentation <https://www.ivoa.net/documents/ADQL/20180112/PR-ADQL-2.1-20180112.html>`__
+ - `LSST Qserv User Guide <https://qserv.lsst.io/user/index.html>`__
+
+
+.. _Adql-Recipes-Explore-Tables:
+
+Exploring tables
+================
+
+When learning about the contents of a table, it can be handy to simply retrieve all columns for "a bunch" (hundreds to thousands) of rows
+and take a look at the results.
+For this use-case, it is recommended to use the ``SELECT TOP`` statement, like in the example below that just retrieves the first 100 rows of the ``Object`` table.
+
+.. code-block:: SQL
+    SELECT TOP 100 * FROM dp02_dc2_catalogs.Object
```

Comment:
For some reason this code block example search using TOP doesn't show up on the documentation repo in the branch https://dp0-2.lsst.io/v/PREOPS-1645/data-access-analysis-tools/adql-recipes.html#general-advice maybe I am missing something about how these are linked? Please disregard if things are as expected here.

Reply:
Thank you! There needs to be a space between the ".. code-block:: SQL" line and the actual code. I added the space and now the code block appears in the rendered document.